README.md

```diff
@@ -1 +1,10 @@
-# kubernetes-leader-election
\ No newline at end of file
+# kubernetes-leader-election
+=======
+## Kubernetes Leader Election
+
+Leader election for kubernetes using leases
```

Comment:
This could be beefed up.  What's a leader and what do you need a leader for?  Maybe examples?  What problem does this solve?

I'm asking for me too :)

Reply:
Yeah, I'll come back through and do that bit last when I'm done fiddling around.
The TLDR is a zero-ish config leader election.

If you are running many pods, and you really want or need one of them to be responsible for preforming some action.

```javascript
if (!elector.current_leader) return
// do important stuff that can only be done by me
```

Think cache manager w/o all the redis coordination and watcher bits.
its pushed off to kubernetes

its a common pattern in kubernetes operators, but all that machinery isn't free unless you're building in Golang and using kubebuilder specifically